portfolio/src/main/webapp/script.js

```diff
@@ -181,6 +184,65 @@ function commentToString(comment) {
         "${comment.text}"`;
 }
 
+/*
+ * Shows comment form and logout prompt if user is logged in,
+ * and shows login prompt if user is not logged in.
+ */
+function showCommentFormAndLoginPrompt() {
+  fetch('/login-status')
+      .then((response) => {
+        return response.json();
+      })
+      .then((loginStatus) => {
+        const isLoggedIn = loginStatus.isLoggedIn;
+        showCommentForm(isLoggedIn);
+        showLoginOrLogoutForm(isLoggedIn);
+      });
+}
+
+function showCommentForm(isLoggedIn) {
+  const commentForm = document.getElementById('comment-form');
+  if (isLoggedIn) {
+    commentForm.classList.remove('hidden');
+  } else {
+    commentForm.classList.add('hidden');
+  }
```

Comment:
You can use Javascript's `.toggle()` for these types of actions

Reply:
I will keep as it as is because say the user refreshes the page, then fetch is called for login for a second time. Here the user is still logged in and we would not want to toggle visibility.